packages/lib/passport-ui/convert-icons.mjs

```diff
@@ -0,0 +1,49 @@
+import camelCase from "camelcase";
+import * as fs from "fs";
+import path from "path";
+import * as prettier from "prettier";
+import svgToUrl from "svg-to-url";
+
+/**
+ * This file converts the icons in the ./icons directory into data URLs, then
+ * creates a src/icons/index.ts file which exports the resulting strings.
+ * This performs the same task that a webpack/esbuild data loader would do, but
+ * without requiring a bundler.
+ * This is helpful because it means we can build this package using only tsc.
+ * It should not be necessary to run this script manually, as it is run
+ * automatically as part of the build process for this package.
+ */
+async function main() {
+  const iconFiles = fs.readdirSync("icons");
+  const newestIconTimestamp = iconFiles.reduce((newest, filename) => {
+    return Math.max(newest, fs.statSync(`icons/${filename}`).mtimeMs);
+  }, 0);
+
+  if (newestIconTimestamp <= fs.statSync("src/icons/index.ts").mtimeMs) {
+    console.log("No changed icons detected");
+    return;
+  }
```

Comment:
Noting that based on my understanding of how this stuff works I can see like some chance this could be buggy. e.g. i've sometimes encountered files that i extract from a zip file or download from the internet that have some sort of old 'creation time' that is not the same time as the time that the file was put into my file system. leaving it to you to decide whether to do something about it / create an issue / implement another solution / whatever else. Not sure how high the cost is to just run this script every build without checking if the file is outdated.

Reply:
I've replaced this with a check to compare the output with the file on-disk, so it will always read the full set of icons and generate the output before comparing it.